spec/meilisearch/client/keys_spec.rb

```diff
@@ -1,60 +1,143 @@
+      expect do
+        new_client.keys
+      end.to raise_meilisearch_api_error_with(403, 'invalid_api_key', 'auth')
+    end
 
-  it 'fails to search if no key used' do
-    new_client = MeiliSearch::Client.new(URL)
-    expect do
-      new_client.index(random_uid).settings
-    end.to raise_meilisearch_api_error_with(401, 'missing_authorization_header', 'auth')
-  end
+    it 'fails to search if no key used' do
+      new_client = MeiliSearch::Client.new(URL)
+      expect do
+        new_client.index(random_uid).settings
+      end.to raise_meilisearch_api_error_with(401, 'missing_authorization_header', 'auth')
+    end
 
-  it 'succeeds to search when using public key' do
-    uid = random_uid
-    public_key = client.keys['public']
-    index = client.create_index(uid)
-    response = index.add_documents(title: 'Test')
-    index.wait_for_pending_update(response['updateId'])
+    it 'succeeds to search when using public key' do
+      uid = random_uid
+      index = client.index(uid)
+      index.add_documents!(title: 'Test')
 
-    new_client = MeiliSearch::Client.new(URL, public_key)
-    response = new_client.index(uid).search('test')
-    expect(response).to have_key('hits')
-  end
+      new_client = MeiliSearch::Client.new(URL, public_key['key'])
+      response = new_client.index(uid).search('test')
+      expect(response).to have_key('hits')
+    end
 
-  it 'succeeds to get settings when using private key' do
-    uid = random_uid
-    client.create_index(uid)
-    private_key = client.keys['private']
-    new_client = MeiliSearch::Client.new(URL, private_key)
-    response = new_client.index(uid).settings
-    expect(response).to have_key('rankingRules')
+    it 'succeeds to get settings when using private key' do
+      uid = random_uid
+      client.create_index!(uid)
+      new_client = MeiliSearch::Client.new(URL, private_key['key'])
+      response = new_client.index(uid).settings
+      expect(response).to have_key('rankingRules')
+    end
   end
 
-  it 'works with method aliases' do
-    expect(client.method(:keys) == client.method(:get_keys)).to be_truthy
+  context 'Test the key managements' do
+    it 'gets the list of the default keys' do
+      results = client.keys['results']
+      expect(results).to be_a(Array)
+      expect(results.count).to be >= 2
+    end
+
+    it 'creates a key' do
+      key_options = {
+        description: 'A new key to add docs',
+        actions: ['documents.add'],
+        indexes: ['*'],
+        expiresAt: nil
+      }
+      new_key = client.create_key(key_options)
+      expect(new_key['expiresAt']).to be_nil
+      expect(new_key['key']).to be_a(String)
+      expect(new_key['createdAt']).to be_a(String)
+      expect(new_key['updatedAt']).to be_a(String)
+      expect(new_key['indexes']).to eq(['*'])
+      expect(new_key['description']).to eq('A new key to add docs')
+    end
+
+    it 'creates a key using snake_case' do
```

Comment:
> Because is the same "test", but we have a major difference, so how to handle that in a good way is the question, we could keep duplicating the tests but soon it will be a problem 😅

I completely agree, this is a real issue. I did not take the time to think about a solution to avoid duplication. The first step should be to open an issue to ask help of the community, by suggesting solutions for example.

Reply:
I created this issue https://github.com/meilisearch/meilisearch-ruby/issues/287 to address the problem in the future :)